src/media/media.ts

```diff
@@ -0,0 +1,32 @@
+import { getCurrentInstance, isRef, onUnmounted, ref, Ref, unref, watch } from "vue"
+
+export function media(query: Ref<string> | string): Ref<boolean> {
+    let mediaQuery = window.matchMedia(unref(query))
+    const matches = ref(mediaQuery.matches)
+    let unwatch
+
+    function updateMatches(event: MediaQueryListEvent) {
+        matches.value = event.matches
+    }
+
+    mediaQuery.addEventListener('change', updateMatches)
+
+    if(isRef(query)) {
+        unwatch = watch(query, () => {
+            mediaQuery.removeEventListener('change', updateMatches)
+            mediaQuery = window.matchMedia(unref(query))
```

Comment:
not sure if intentional, but it seems like 
 - you add an event listener on 'change' line 12
 - then if `query` is a ref you add a watcher
 - once that watch fires at least once you remove the event listener from 12
 - you do NOT re-add the event listener on 'change'

Reply:
Great catch. Fixed